bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/UpdateSecretCommand.cs

```diff
@@ -31,12 +31,12 @@ public async Task<Secret> UpdateAsync(Secret updatedSecret, Guid userId)
         var orgAdmin = await _currentContext.OrganizationAdmin(secret.OrganizationId);
         var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
 
-        var project = updatedSecret.Projects?.FirstOrDefault();
+        var originalSecretsProject = secret.Projects?.FirstOrDefault();
 
         var hasAccess = accessClient switch
         {
             AccessClientType.NoAccessCheck => true,
-            AccessClientType.User => project != null && await _projectRepository.UserHasWriteAccessToProject(project.Id, userId),
+            AccessClientType.User => originalSecretsProject != null && await _projectRepository.UserHasWriteAccessToProject(originalSecretsProject.Id, userId),
```

Comment:
This used to check if the project you are updating it TO has write permissions, we want to check if the project it WAS assigned to before you tried updating it has Write permission.

Reply:
i know that they won't get the project in the dropdown if they don't have write but it feels like we should still check for permission on the target project?

Reply:
Even though the UI should handle the situation you are mentioning, it would be beneficial to add a check for that as well, updating the PR now 😄 thanks Patrick!